src/config.rs

```diff
@@ -0,0 +1,136 @@
+//! Provides a simple way of reading configuration files.
+//!
+//! Reads a TOML config file for the IPC Agent and deserializes it in a type-safe way into a
+//! [`Config`] struct. Usage:
+//!
+//! ```
+//! let c: Config = Config::from_file("config/example.toml");
```

Comment:
We can put this in doc tests are well.

Reply:
Opted to remove since this example become obsolete with the removal of the file from the source tree.